server/src/main/java/org/opensearch/gateway/remote/RemoteObject.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.gateway.remote;
+
+import org.opensearch.common.CheckedRunnable;
+import org.opensearch.core.action.ActionListener;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * An interface to read/write and object from/to a remote storage. This interface is agnostic of the remote storage type.
+ * @param <T> The object type which can be upload to or download from remote storage.
+ */
+public interface RemoteObject<T> {
+    public T get();
+
+    public String clusterUUID();
```

Comment:
why did we need this? lets figure out a different way to wire this? In the constructor for AbstractRemoteObject (a new class)?

Reply:
This is need to construct the blob path of the remote object. I can move it down to AbstractRemoteBlobStoreObject.